### ISSUE TITLE
Add support for JIRA issue tracker

### ISSUE THREAD

**Issue by thc202** [zaproxy/zaproxy#2990]
Extending from #440, it should be possible to raise the issues in a JIRA issue tracker. [1]

Add-on:
Bug Tracker

[1] https://www.atlassian.com/software/jira

**Comment by kerberosmansour** [comment]
@thc202 do you see this done in May or April?

**Comment by kingthorin** [comment]
If someone decides to tackle it :-) All PRs welcome.

https://github.com/zaproxy/zap-extensions/tree/alpha/src/org/zaproxy/zap/extension/bugtracker

**Comment by psiinon** [comment]
This could be a good student project...

**Comment by thc202** [comment]
@kerberosmansour since it does not depend on me (not assigned to me, just raised the issue) it's hard to predict...

**Comment by thc202** [comment]
#2991 might be a good addition to the project.

**Comment by nhegde610** [comment]
Hi, I would love to take this project as my GSOC project. Is anyone working on this?

**Comment by omerlh** [comment]
This is already supported when using [OWASP Glue](https://github.com/owasp/glue) to run Zap, including filtering of existing items and many other cool features. What do you think?

**Comment by omerlh** [comment]
I was trying to say that the benefit of Glue is the fact you can do such things once and enjoy them all over the place, with any security tool. Also, there are many other CI utilities (alert filtering, TeamCity reporting etc) - and it might not be right to add all of them to Zap.

**Comment by psiinon** [comment]
I have no problem with that :)
Do you know how many installations Glue has? I suspect that the vast majority of ZAP users wont be using it right now, but I could be wrong.

**Comment by thc202** [comment]
This would also be used by the Jenkins plugin (currently relying on 3rd party add-on, unmaintained I think).

**Comment by omerlh** [comment]
@psiinon that the main problem with Glue right now - the project seemed unmaintained, I maintain our own fork (and not happy to do that) and trying to gain some popularity to it as I think it could help the community with many things. It could also maybe replace the Jenkins plugin if it is unmaintained (as @thc202 mentioned). Currently, Jenkins is not supported - but this could be added easily I guess.

**Comment by psiinon** [comment]
Thats a shame. I dont really want to recommend that people should use an unmaintained project. All the more reason to support it within ZAP then :)

**Comment by omerlh** [comment]
Well, the fork is maintained - but I agree with you. I just think that Glue can help with so many things, and this is the reason I keep pushing it. I also believe that part of the fact it is unmaintained is related to the lack of users - it's a chicken and egg situation...

**Comment by psiinon** [comment]
Maintaining projects is hard work, as you know ;)
However it looks like Glue is the second most popular OWASP project on dockerhub: https://hub.docker.com/u/owasp/dashboard/ so it might be getting a fair amount of use.

**Comment by omerlh** [comment]
Yep, I've also noticed that. It already supports many good security tools (Snyk, Checkmarx, and others) so this might be the reason.
On the other hand, no new issues for last half year besides mine...
